src/component/button/button.scss

```diff
@@ -56,6 +56,26 @@
       }
     }
 
+    &.catalog {
```

Comment:
The `button` component is a generic component that should not contain any "catalog" tinted class or other but only `primary`, `secondary` or other classes being compliant with the design system. 
You can check on the design system that the css code you add is the one corresponding to the `primary` variant button. You should change this to keep the `button` generic ;)